### PR TITLE
[CDAP-13956] [CDAP-13971] Fixes kafka connection to pass in right schema to pipeline

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Fixes schema passed on from dataprep to pipelines when creating a pipeline out of a dataprep table
- Fixes S3 to not incorrectly truncate bucket name while reading a file from s3
- Fixes kafka connection to show right content on repeated clicks on the connection in left panel
- Fixes bigquery to avoid redundant calls while fetching tables under a dataset vs just list of datasets
- Fixes breadcrumbs to behave consistently for GCS and S3 browsers when operating from pipeline studio
- Adds a reset state to dataprep browser store to start from scratch when coming again to the browser in pipeline studio (we used to set `loading` to any connection to `true` after reading a file. So going back the user wouldn't be able to open the connection as we check for `loading` to be false to avoid redundant backend calls 😭 )

JIRA:
 https://issues.cask.co/browse/CDAP-13971
 https://issues.cask.co/browse/CDAP-13956
Build: https://builds.cask.co/browse/CDAP-UDUT25

### CODE REVIEW

File: cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/DataPrepBrowserStore/Actions/bigquery.js [hunk 1]
```diff
@@ -23,13 +23,48 @@ import {objectQuery} from 'services/helpers';
...
+      }}
+    />
+    Here in the render method we set bigquery as the default browser which will again call this
+    7. Without this check we get the list of datasets AGAIN even though the user clicked on A dataset
+    8. In the /connections/bigquery/:bigqueryid/datasets/:datasetid which renders TableList makes the list of tables call
+    9. This is redundant.
+
+    This will prevent the user from clicking on the connection in the left panel to refresh the contents.
+  */
+  if (bigquery.loading || bigquery.datasetList.length) { return; }
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/DataPrepBrowserStore/Actions/bigquery.js]: does this only happen to bigquery? that seemed odd.

Also won't this prevent changing to a different bigquery connections if the datasetList length is the same?

File: cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/S3Browser/index.js [hunk 2]
```diff
@@ -52,7 +52,11 @@ export default class S3Browser extends Component {
     const {selectedNamespace: namespace} = NamespaceStore.getState();
     const {connectionId, prefix} = DataPrepBrowserStore.getState().s3;
     const strippedPrefix = prefix.slice(prefix.indexOf('/') + 1);
-    const activeBucket = strippedPrefix.slice(0, strippedPrefix.indexOf('/'));
+    const trailingSlashIndex = strippedPrefix.indexOf('/');
+    let activeBucket = strippedPrefix;
+    if (trailingSlashIndex !== -1) {
+      activeBucket = strippedPrefix.slice(0, trailingSlashIndex);
+    }
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/S3Browser/index.js]: is this trying to remove the ending `/` on a path? If so, wouldn't it be better to just check the last character if it's a `/` or not?

File: cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/DataPrepBrowserStore/Actions/bigquery.js (hunk 1 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/DataPrepBrowserStore/Actions/bigquery.js]: Good catch.

**Review by elfenheart — APPROVED** [review]
👍  LGTM

**Review by bdmogal — APPROVED** [review]

**Pull request merged**